src/common/annotationsFromFrame.ts

```diff
@@ -0,0 +1,132 @@
+import { AnnotationEvent, DataFrame, FieldType, Field, KeyValue } from '@grafana/data';
+
+interface AnnotationEventNames {
+  userId?: string;
+  login?: string;
+  email?: string;
+  avatarUrl?: string;
+  time?: string; // or first time
+  timeEnd?: string;
+  title?: string;
+  text?: string; // or first string
+  tags?: string; // Split on ','
+}
+
+const defaultAnnotationEventFinder: AnnotationEventNames = {
+  userId: 'userId',
+  login: 'login',
+  email: 'email',
+  avatarUrl: 'avatarUrl',
+  time: 'time',
+  timeEnd: 'timeEnd',
+  title: 'title',
+  text: 'text',
+  tags: 'tags',
+};
+
+interface AnnotationEventFieldSetter {
+  key: string;
+  field: Field;
+
+  regex?: RegExp; // for text
+  split?: string; // for tags
+}
+
+export interface AnnotationsFromFrameOptions {
+  field?: AnnotationEventNames;
+  regex?: AnnotationEventNames;
+}
+
+export function getAnnotationsFromFrame(frame: DataFrame, options?: AnnotationsFromFrameOptions): AnnotationEvent[] {
```

Comment:
FYI!  I am trying to add this to core grafana/runtime:

https://github.com/grafana/grafana/pull/26634/files

so don't work too much here! or make sure to copy that version :)